draft-ietf-mls-protocol.md

```diff
@@ -3133,6 +3163,31 @@ set to `key_package`.
 Extension included in the `extensions` or `leaf_node.extensions` fields MUST be
 included in the `leaf_node.capabilities` field.
 
+## Directed KeyPackages
+
+Before a new member can be added to the group, the member that proposes their
+addition needs to get a KeyPackage for the new member. As discussed in
+{{protocol-overview}}, this can be done either by pre-publishing KeyPackages
+to a directory or by having the joining member send them proactively.
+
+In the latter case, the sender of the KeyPackage can limit the scope of the
+KeyPackage by including an extension of type `target_group_id`, of the following
+form:
+
+```
+opaque group_id<V>;
+```
+
+As defined in {{keypackage-validation}}, when this extension is present, the
+members of the group will verify that the KeyPackage is intended for the group.
+
+In applications where both pre-published and directed KeyPackage distribution
+are supported, the application SHOULD require that this extension be present in
+a KeyPackage sent proactively by a joiner, and absent in pre-published
+KeyPackages. This prevents pre-published KeyPackages from being fetched by a
+third party and used to request membership in a group (since the pre-published
+KeyPackages will not have this extension).
```

Comment:
I don't understand the value of having this extension. If pre-published KeyPackages already exist without this extension, then I can just use them to do whatever nefarious thing you're trying to stop from happening with the directed KeyPackages.